components/script/dom/webidls/GPUCanvasContext.webidl

```diff
@@ -0,0 +1,9 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+
+[Exposed=(Window, DedicatedWorker), Pref="dom.webgpu.enabled"]
+interface GPUCanvasContext {
+    readonly attribute (HTMLCanvasElement or OffscreenCanvas) canvas;
+};
```

Comment:
`./mach build --no-default-features` works.

Building with `--feature webgpu` fails right now because of this change. It's because now we have `GPUCanvasContext` at two different location when running with `webgpu`(see `partial interface GPUCanvasContext in webgpu.webidl). I need to fix this

Reply:
so to resolve this issue I added `Skip-if CARGO_FEATURE_WEBGPU` in [b8054a6](https://github.com/servo/servo/pull/34444/commits/b8054a6c3c0d7cb2607740a66e271747fc0740d0).  Before doing this, I tried a few options like having partial interface, base Interface with Inheritance or renaming the dummy interface, but it didn't work and felt like I was going in circle.

Tbh, I am not happy with skip-if option either, as we add this just for this one case and that's not optimal. I am open to other ideas. Is there a way to just skip this? Looks like having GPUCanvasContext is needed in both cases(with or without webgpu)

cc @sagudev

Reply:
I think pref guards might be causing problems. I didn't try it but something like this should work:
```idl
[Exposed=(Window, DedicatedWorker)]
interface GPUCanvasContext {
    readonly attribute (HTMLCanvasElement or OffscreenCanvas) canvas;
};

partial interface GPUCanvasContext {
    [Throws, Pref="dom.webgpu.enabled"]
    undefined configure(GPUCanvasConfiguration descriptor);
    [Pref="dom.webgpu.enabled"] undefined unconfigure();
    [Throws, Pref="dom.webgpu.enabled"]
    GPUTexture getCurrentTexture();
};
```